pkg/internal/apis/config/validate.go

```diff
@@ -130,6 +140,68 @@ func (n *Node) Validate() error {
 	return nil
 }
 
+func formatPortMapping(addr net.IP, port int, protocol PortMappingProtocol) string {
+	return fmt.Sprintf("%s/%s",
+		net.JoinHostPort(addr.String(), strconv.Itoa(port)),
+		protocol,
+	)
+}
+
+func validatePortMapping(portMappingsSet sets.String, addrSet sets.String, mapping PortMapping) error {
+	wildcardAddrIPv4 := net.ParseIP("0.0.0.0")
+	wildcardAddrIPv6 := net.ParseIP("::")
+
+	addr := net.ParseIP(mapping.ListenAddress)
+	port := int(mapping.HostPort)
+	protocol := mapping.Protocol
+	ipFamily := getIPFamily(addr)
+
+	portMappingString := formatPortMapping(addr, port, protocol)
```

Comment:
so, you have two dimensions "port-protocol" and then address, if you aggregate first by port protocol this seems it will be more simplers

key = port-protocol
and you create a map with key the port protocol and values the listen addresses
map[key][]string
then if there is a wildcard in the value and len(string) > 1 you have a collision
if there are 2 values with the same address in the string then you have a collision

I think you can also use a set instead of the string slice to use the existing methods to detect the collisions while processing these structures